bugsnag-android-core/src/test/java/com/bugsnag/android/JournaledDocumentTest.kt

```diff
@@ -48,9 +56,15 @@ class JournaledDocumentTest {
     fun testEmptyDocumentNonEmptyJournal() {
         val baseDocumentPath = folder.newFile("mydocument")
```

Comment:
I wasn't able to get this to work as a class member. It can't be initialized at object instantiation because `newFile()` would throw `java.lang.IllegalStateException: the temporary folder has not yet been created`.

Reply:
Could `File.createTempFile` do the job?

Reply:
I started off using `File.createTempFile` but ran into permissions problems (it kept saying permission denied), which is why I switched to `TemporaryFolder`. `TemporaryFolder` also has the feature of auto-cleanup so that tests don't leave cruft behind even if the test fails or exceptions out.